### PR TITLE
Change session timeout technique

### LINKED ISSUE #730: Use a diffrent type of session cookie?
I have a little annoying problem here, I run a few different services on the same box where MailWatch runs, so every time I log out of MailWatch I also get log out of my other Web interface and vice versa There must be some kind of php session conflict. 

Perhaps this could be a source for many other problems, which relay on php session/tokens and so on?
```

FreeBSD 10.3-RELEASE-p17
MailWatch Version 1.2.0
MailScanner Version 4.85.2
ClamAV Version 0.99.2
SpamAssassin Version 3.4.1
PHP Version 5.6.30
MySQL Version 10.1.21-MariaDB
NGINX 1.10.2
```

### PR DESCRIPTION
**Pull request by shawniverson**
Fixes #730

### CODE REVIEW

**Comment by shawniverson** [comment]
Hold on merge, testing.

File: mailscanner/logout.php [hunk 1]
```diff
@@ -51,4 +51,8 @@
 
 session_destroy();
 
-header('Location: index.php');
+if (isset($_GET['error'])) {
+    header('Location: login.php?error=' . $_GET['error']);
```
Comment (endelwar) [inline mailscanner/logout.php]: filter this $_GET fox xss

File: mailscanner/functions.php [hunk 2]
```diff
@@ -287,6 +286,12 @@ function html_start($title, $refresh = 0, $cacheable = true, $report = false)
         }
     }
 
+    if (!isset($_SESSION['last_update']) || $_SESSION['last_update'] + SESSION_TIMEOUT < time()) {
```
Comment (Skywalker-11) [inline mailscanner/functions.php]: This would always be true because the last_update has already been overwritten in line 127

**Comment by shawniverson** [comment]
Yeah, I have to rethink this logic...
On Apr 11, 2017 6:23 AM, "Michael Pietsch" <notifications@github.com> wrote:

*@Skywalker-11* commented on this pull request.
------------------------------

In mailscanner/functions.php
<https://github.com/mailwatch/MailWatch/pull/744#discussion_r110866204>:

> @@ -287,6 +286,12 @@ function html_start($title, $refresh = 0, $cacheable = true, $report = false)
         }
     }

+    if (!isset($_SESSION['last_update']) || $_SESSION['last_update']
+ SESSION_TIMEOUT < time()) {

This would always be true because the last_update has already been
overwritten in line 127

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/mailwatch/MailWatch/pull/744#pullrequestreview-32066821>,
or mute the thread

File: mailscanner/functions.php (hunk 2 above)
Comment (shawniverson) [inline mailscanner/functions.php]: Moved line 127 to checklogin.php

**Comment by shawniverson** [comment]
Ready to test some more...

**Comment by shawniverson** [comment]
Adjusting values to test for a timing issue...

**Comment by stefaweb** [comment]
StyleCI

**Comment by shawniverson** [comment]
@stefaweb I don't agree with StyleCl's anysis here.  It is getting thrown off by the HTML/PHP mix, I think...

**Comment by shawniverson** [comment]
@endelwar After testing this with the help of @spec1re, we are seeing inconsistent results at 5 minutes for the session timeout.  However, at 10 minutes, we are both able to get consistent results.  Seems browser related.  The PR is ready for your review.

**Comment by endelwar** [comment]
Looks good to me, I agree on StyleCI gone crazy. Merging right now.

**Comment by Skywalker-11** [comment]
Is there a way to exclude the file from styleci?

**Comment by endelwar** [comment]
@Skywalker-11 I've just written to the StyleCI developer asking for correctness of the analisys

**Comment by Skywalker-11** [comment]
@endelwar any news on this?

**Comment by endelwar** [comment]
I'm reporting here the reply, but I don't agree with the "view" issue:
```
Hi Manuel,

Thanks for getting in touch. StyleCI is not designed to fix "view" files. It treats them as source code, so, the indentation change there is correct, it just looks funny, because it's in a view file.

I'd recommend excluding your view files from analyses.
```

We should refactor login.php, maybe by indenting better the php code part or convert html to php strings.

**Pull request merged**